zrml/market-commons/src/migrations.rs

```diff
@@ -16,60 +16,481 @@
 // You should have received a copy of the GNU General Public License
 // along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
 
-// We use these utilities to prevent having to make the swaps pallet a dependency of
-// prediciton-markets. The calls are based on the implementation of `StorageVersion`, found here:
-// https://github.com/paritytech/substrate/blob/bc7a1e6c19aec92bfa247d8ca68ec63e07061032/frame/support/src/traits/metadata.rs#L168-L230
-// and previous migrations.
-mod utility {
-    use alloc::vec::Vec;
-    use frame_support::{
-        storage::{storage_prefix, unhashed},
-        traits::StorageVersion,
-        StorageHasher,
-    };
-    use parity_scale_codec::Encode;
+use crate::{
+    AccountIdOf, AssetOf, BalanceOf, BlockNumberOf, Config, MarketIdOf, MomentOf,
+    Pallet as MarketCommons,
+};
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+use frame_support::{
+    log,
+    pallet_prelude::{Blake2_128Concat, StorageVersion, Weight},
+    traits::{Get, OnRuntimeUpgrade},
+};
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+use sp_runtime::{Perbill, RuntimeDebug, Saturating};
+use zeitgeist_primitives::types::{
+    Deadlines, EarlyClose, Market, MarketBonds, MarketCreation, MarketDisputeMechanism,
+    MarketPeriod, MarketStatus, MarketType, OutcomeReport, Report, ScoringRule,
+};
 
-    #[allow(unused)]
-    pub fn storage_prefix_of_market_common_pallet() -> [u8; 32] {
-        storage_prefix(b"MarketCommons", b":__STORAGE_VERSION__:")
+cfg_if::cfg_if! {
+    if #[cfg(feature = "try-runtime")] {
+        use alloc::collections::BTreeMap;
+        use frame_support::migration::storage_key_iter;
+        use zeitgeist_primitives::types::MarketId;
     }
+}
 
-    #[allow(unused)]
-    pub fn get_on_chain_storage_version_of_market_commons_pallet() -> StorageVersion {
-        let key = storage_prefix_of_market_common_pallet();
-        unhashed::get_or_default(&key)
+cfg_if::cfg_if! {
+    if #[cfg(any(feature = "try-runtime", test))] {
+        const MARKET_COMMONS: &[u8] = b"MarketCommons";
+        const MARKETS: &[u8] = b"Markets";
     }
+}
```

Comment:
```suggestion
#[cfg(any(feature = "try-runtime", test))]
const MARKET_COMMONS: &[u8] = b"MarketCommons";
#[cfg(any(feature = "try-runtime", test))]
const MARKETS: &[u8] = b"Markets";  
```

Reply:
That's fine, but how many imports until copy-pasting `#[cfg(any(feature = "try-runtime", test))]` is no longer acceptable?`

Reply:
I suggest we use `cfg_if` whenever it results in less lines of code.